tools/arcilator/arcilator-runtime.h

```diff
@@ -1,12 +1,63 @@
 // NOLINTBEGIN
 #pragma once
 #include <array>
+#include <cstdarg>
 #include <cstdint>
+#include <cstdio>
 #include <cstring>
 #include <functional>
 #include <ostream>
 #include <vector>
 
+// Sanity checks for binary compatibility
+#ifdef __BYTE_ORDER__
+#if (__BYTE_ORDER__ != __ORDER_LITTLE_ENDIAN__)
+#error Unsupported endianess
+#endif
+#endif
+static_assert(sizeof(int) == 4, "Unsupported ABI");
+static_assert(sizeof(long long) == 8, "Unsupported ABI");
+
+// ---  Exports to the IR ---
+
+#ifdef _WIN32
+#define ARCEXPORT(rtype) extern "C" __declspec(dllexport) rtype __cdecl
```

Comment:
Can you make this consistent with `MLIR_RUNNERUTILS_EXPORT` macro?

Reply:
`MLIR_RUNNERUTILS_EXPORT` does not include `extern "C"` yet it looks like it is prefixed with `extern "C"` every time it is used. That seems odd to me.
The `__cdecl` is probably not required. It don't have access to my Windows PC right now, I'll check later today. Basically, I've just used every magic incantation here that I could come up with of to increase the likelihood of it linking. 😅